File: POO/tad Lista/java/Stack.java
public class Stack {
    private List list = new List();

    public void push(int value) {
        list.insert(value);
    }

    public void pop() {
        if (list.getSize() > 0) {
            list.removeLast();
        } else {
            System.out.println("Pilha Vazia!");
        }
    }

    public int size() {
        return list.getSize();
    }

    public int top() {
        return list.lastItem();
    }
}

File: POO/tad Lista/c++/stack.cpp
#include <iostream>
#include "stack.h"

void Stack::push(int value)
{
    list.insert(value); // Inserção no início da pilha
}

void Stack::pop()
{
    if (list.getSize() > 0)
    {
        list.removeLast(); // Remoção do início da pilha
    }
    else
    {
        std::cout << "Pilha Vazia!" << std::endl;
    }
}

int Stack::size()
{
    return list.getSize();
}

int Stack::top()
{
    return list.lastItem(); // Retorna o elemento no topo da pilha
}
